odoo/fields.py

```diff
@@ -3441,22 +3441,19 @@ def _add_display_name(cls, values_list, env, value_keys=('value', 'default')):
                         property_definition[value_key] = (property_value, display_name)
                     except AccessError:
                         # protect from access error message, show an empty name
-                        property_definition[value_key] = (property_value, _("No Access"))
+                        property_definition[value_key] = (property_value, None)
                     except MissingError:
                         property_definition[value_key] = None
 
                 elif property_type == 'many2many' and property_value and is_list_of(property_value, int):
                     try:
-                        display_names = env[property_model].browse(property_value).mapped('display_name')
-                        property_definition[value_key] = [
-                            (record_id, display_name)
-                            for record_id, display_name, in zip(property_value, display_names)
-                        ]
-                    except AccessError:
-                        # protect from access error message, show an empty name
+                        has_access = env[property_model].check_access_rights('read', raise_exception=False)
```

Comment:
`records._filter_access_rules('read')` is not enough, we can have `display_name` depending of other models.

Maybe something like that instead :
```python
property_definition[value_key] = []
for record in records:
    try:
        property_definition[value_key].append(record.id, record.display_name)
    except AccessError:
        property_definition[value_key].append(record.id, None)
```

Reply:
indeed good catch :) I added a test with new access rules